nix-repl.el

```diff
@@ -34,10 +34,30 @@
     (define-key map "\t" 'completion-at-point)
     map))
 
+(defun nix-repl-save-all-histories ()
+  "Call `comint-write-input-ring' for all `nix-repl-mode' buffers."
+  (dolist (buf (buffer-list))
+    (with-current-buffer buf
+      (when (eq major-mode 'nix-repl-mode)
+        (comint-write-input-ring)))))
+
 (define-derived-mode nix-repl-mode comint-mode "Nix-REPL"
   "Interactive prompt for Nix."
   (setq-local comint-prompt-regexp nix-prompt-regexp)
   (setq-local comint-prompt-read-only t)
+  (let* ((is-remote (file-remote-p default-directory))
+         (maybe-xdg-data-home (shell-command-to-string "echo -n $XDG_DATA_HOME"))
```

Comment:
use `getenv` to avoid a subprocess

Reply:
OK, I used `getenv` for non-remote. I don't really know how things work in remote settings; I saw that Emacs' `shell.el` used a subprocess for remote settings and `getenv` for local settings:

``` elisp
(when remote
        ;; `shell-snarf-envar' does not work trustworthy.
        (setq hsize (shell-command-to-string "echo -n $HISTSIZE")
              hfile (shell-command-to-string "echo -n $HISTFILE")))
```

So I updated this patch to do the same thing. Is that the right way of doing this?